### DISCUSSION TITLE
Some ideas as a blogging engine

### DISCUSSION THREAD

**Discussion by sumy7** [web-infra-dev/rspress#381] [Ideas]
最近考虑将博客从hexo迁移到rspress，有几个优化的地方是否可以考虑支持一下？

1. 我发现页面frontmatter可以配置pageType，但是只有固定的几类，能否根据pageType自动调用theme里导出的同名layout。用于扩展支持自定义布局。
2. plugin在extendPageData方法里修改pageData.frontmatter是无效的。场景是想默认禁用所有页面的【侧边栏】。不知道还有没有其它方法？
3. 【上一篇】或者【下一篇】的数据能否使用pageData里的字段，现在用的是【侧边栏】的数据，通过plugin的addPage添加的页面就无法展示这两个按钮了。
4. 希望能够将页面组件的粒度继续拆分一下：比如lastUpdate和prevPageNextPage组件，这样编写自定义布局的时候更方便的重复利用组件
5. 还有最重要的一点，还希望layout能够多点扩展槽，比如H1标题之后，docFooter之后

当然这些内容通过自定义布局的模式都能实现，但是还是希望能多一些扩展点。

**Comment by sanyuan0704** [comment]
很好的建议！未来 Rspress 会以如下的方向进一步提升扩展性：
- 支持自定义 Layout。根据 pageType  自动渲染对应的 Layout 组件。
- 插件钩子支持修改文档的 frontmatter 和 content。
- 主题导出 LastUpdate 和 PrevNextPage 组件。
- 增加主题插槽。如果有特殊需求，也欢迎来提 PR 哈。

另外，判断上一篇、下一篇的逻辑，一般是和侧边栏的顺序是对应的，pageData 里面的排序并不能对应这一顺序，如果你有 addPage 增加的自定义页面，建议在侧边栏注册自定义页面的路由，这样两个按钮就可以出现了。

**Comment by Liangdi** [comment]
最近也正在进行 rspress 博客改造，看了题主的 blog 收益匪浅， rspress 目前确实在 theme 方面确实需要添加一些灵活性。
我觉得可以增加独立 theme 目录的配置，做到整体替换 theme , 同时完善主题扩展机制和文档, 这样可以吸引一些其他主题的移植

**Reply by Timeless0911** [reply]
感谢你的建议，我们将在 Rspress V2 中完善主题的扩展和继承。
